app/celery/tasks.py

```diff
@@ -486,8 +488,21 @@ def persist_daily_sorted_letter_counts(day, file_name, sorted_letter_counts):
     dao_create_or_update_daily_sorted_letter(daily_letter_count)
 
 
+@dataclass
+class NotificationUpdate:
+    """
+    A NotificationUpdate is used to wrap a row of a DVLA response file.
+    `despatch_date` is optional to support files which contain it and those that don't
+    """
+
+    reference: str
+    status: str
+    page_count: str
+    cost_threshold: str
+    despatch_date: Optional[str] = None
```

Comment:
I wonder if it might be nice to take the opportunity to have `page_count` and `despatch_date` as native objects (int and date) now? Possibly that means a few more changes to how they get used elsewhere - up to you whether it's worth it.

Reply:
I think `despatch_date` I'd prefer to leave for now - there will be an upcoming story to start using that field which might be a good time to change how it works, but I'd like to just make sure we are receiving the field in the expected format with no errors before trying to parse it or change its type.

Were you thinking that `page_count` could get changed to in an `int` in the dataclass, or before it's passed to it?

Reply:
So this is a bit of a quirk of dataclasses. If you define page_count as an int with `page_count: int`, it won't actually do any mutation of data as the instance is being made. If you pass in `"2"` to the field, you will end up with a string still. So you need to explicitly convert it to an integer before/as you pass it into the constructor.

So the line `notification_updates = [NotificationUpdate(*line.split("|")) for line in response_file.splitlines()]` would need to pull out the `page_count` and convert it to an int explicitly. It would be nice/more correct to use the right data types for each thing, but it may not be worth it right now as so if you want to keep it as-is, I'm OK with it.

Reply:
Thanks, I knew it wouldn't happen automatically in the dataclass but wasn't sure if you were thinking of [a solution like this](https://stackoverflow.com/a/54863771). I think I'd probably keep things as-is for now mainly because it keeps the code a little simpler and once we're fully migrated to sending letters via the API we may stop using response files.